shell/platform/darwin/macos/framework/Source/FlutterOpenGLRenderer.h

```diff
@@ -0,0 +1,86 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import <Cocoa/Cocoa.h>
+
+#import "flutter/shell/platform/darwin/macos/framework/Headers/FlutterEngine.h"
+#import "flutter/shell/platform/darwin/macos/framework/Source/FlutterView.h"
+#import "flutter/shell/platform/embedder/embedder.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ * Provides the renderer config needed to initialize the embedder engine and also handles external
+ * texture management. This is initialized during FlutterEngine creation and then attached to the
+ * FlutterView once the FlutterViewController is initializer.
+ */
+@interface FlutterOpenGLRenderer : NSObject <FlutterTextureRegistry>
+
+/**
+ * The resource context used by the engine for texture uploads. FlutterViews associated with this
+ * engine should be created to share with this context.
+ */
+@property(nonatomic, readonly, nullable) NSOpenGLContext* resourceContext;
+
+/**
+ * The main OpenGL which will be used for rendering contents to the FlutterView.
+ */
+@property(readwrite, nonatomic, nonnull) NSOpenGLContext* openGLContext;
+
+/**
+ * Intializes the renderer with the given FlutterEngine.
+ */
+- (instancetype)initWithEmbedderEngine:(FLUTTER_API_SYMBOL(FlutterEngine))engine
+                         flutterEngine:(FlutterEngine*)flutterEngine;
+
+/**
+ * Attaches to the FlutterView and sets up the renderers main OpenGL context.
+ */
+- (void)attachToFlutterView:(FlutterView*)view;
+
+/**
+ * Called by the engine to make the context the engine should draw into current.
+ */
+- (bool)makeCurrent;
```

Comment:
These can all be BOOLs as well, and the implementations updated to YES/NO. (I should really have done that before when the statics were updated to call directly to ObjC versions; this is a good time to fix it.)

Reply:
Done